### DISCUSSION TITLE
数据绑定问题

### DISCUSSION THREAD

**Discussion by NameofZY** [BYJRK/DotNet-Discussions#210] [Q&A]
<ItemsControl ItemsSource="{Binding Equipments}"
              Padding="5">
    <i:Interaction.Triggers>
        <i:EventTrigger EventName="MouseLeftButtonDown">

            <i:InvokeCommandAction Command="{Binding OpenCommand}"
                                   CommandParameter="{怎样绑定到ItemTemplate中TextBlock的Text？？请指教}" />
        </i:EventTrigger>
    </i:Interaction.Triggers>
    <ItemsControl.ItemsPanel>
        <ItemsPanelTemplate>
            <UniformGrid />
        </ItemsPanelTemplate>
    </ItemsControl.ItemsPanel>

    <ItemsControl.ItemTemplate>
        <DataTemplate>

            <Border BorderBrush="#9918AABD"
                    BorderThickness="0,1,0,0"
                    Background="#0A18AABD"
                    Margin="5">

                <StackPanel>
                    <TextBlock Text="{Binding Title}"
                               FontSize="12"
                               Foreground="White"
                               HorizontalAlignment="Center"
                               Margin="0 15 " />
                    <c:CircularProgressBar Value="{Binding OpenValue}"
                                           Width="80"
                                           Height="80" />
                </StackPanel>
            </Border>
            <DataTemplate.Triggers>
                <Trigger Property="IsMouseOver"
                         Value="true">
                    <Setter Property="Cursor"
                            Value="Hand" />
                </Trigger>
            </DataTemplate.Triggers>
        </DataTemplate>

    </ItemsControl.ItemTemplate>
</ItemsControl>

**Comment by SlimeNull** [comment]
这是很典型的集合绑定问题。

你应该把Behavior写在ItemTemplate中，然后想办法把 Command 绑定到你写好的 Command，关于这个，你可以做一个简单的绑定代理，在窗体资源中创建任何一个 Freezable，然后为该资源的某个属性添加绑定，这样在你的列表中，你就可以通过 StaticResource 引用到这个资源，进而继续绑定到窗口的相关数据或指令。

一般这个 Freezable 是手动实现的一个简单类型，它只包含一个 Data 依赖属性，然后，在你的集合中，绑定语法的 Source 设为这个 StaticResource 就可以了。

（用手机编辑的回复，所以不太方便写代码，如果你需要更详细的解释，回复，我应该会收到电子邮件通知的）
